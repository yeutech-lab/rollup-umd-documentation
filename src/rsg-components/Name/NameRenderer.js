import React from 'react';
import PropTypes from 'prop-types';
import Code from 'rsg-components/Code';
import Styled from 'rsg-components/Styled';
import cx from 'classnames';

export const styles = ({ fontSize, color }) => ({
	name: {
		fontSize: fontSize.small,
		color: color.name,
	},
	isDeprecated: {
		color: color.light,
		textDecoration: 'line-through',
	},
});

export function NameRenderer({ classes, children, deprecated }) {
	const classNames = cx(classes.name, {
		[classes.isDeprecated]: deprecated,
	});
	return (
		<span className={classNames}>
			<Code>{children}</Code>
		</span>
	);
}

NameRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node.isRequired,
	deprecated: PropTypes.bool,
};

export default Styled(styles)(NameRenderer);
