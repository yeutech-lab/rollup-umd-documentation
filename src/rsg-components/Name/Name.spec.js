import React from 'react';
import { NameRenderer, styles } from './NameRenderer';

const props = {
	classes: classes(styles),
};

it('renderer should render argument name', () => {
	const actual = shallow(<NameRenderer {...props}>Foo</NameRenderer>);

	expect(actual).toMatchSnapshot();
});

it('renderer should render deprecated argument name', () => {
	const actual = shallow(
		<NameRenderer {...props} deprecated>
			Foo
		</NameRenderer>
	);

	expect(actual).toMatchSnapshot();
});
