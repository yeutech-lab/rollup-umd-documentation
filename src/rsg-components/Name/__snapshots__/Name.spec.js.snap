// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render argument name 1`] = `
<span
  className="name"
>
  <Styled(Code)>
    Foo
  </Styled(Code)>
</span>
`;

exports[`renderer should render deprecated argument name 1`] = `
<span
  className="name isDeprecated"
>
  <Styled(Code)>
    Foo
  </Styled(Code)>
</span>
`;
